### PR TITLE
Adds Validator Set Additions to the log output

### PR DESCRIPTION
**Pull request by lavabyrd**
@lukanus , as discussed, this adds set additions to the DMLOG output. I didn't want to just merge it incase you haven't anything hardcoded further up the chain for the demo that isn't ready for this. If you're happy to just merge it, go ahead (or let me know and I will).

Cheers

### CODE REVIEW

File: extractor.go [hunk 1]
```diff
@@ -138,47 +138,54 @@ func (ex *ExtractorService) OnStop() {
...
+			setUpdates := valSetMsg.Data().(types.EventDataValidatorSetUpdates)
 
-		// ex.Logger.Info("Validator Set Update", "validator", validator)
+			if err := indexValSetUpdates(w, sync, &setUpdates); err != nil {
+				ex.Logger.Error("failed to index Validator Set Data", "err", err)
+			} else {
+				ex.Logger.Info("indexed validator set updates", setUpdates)
+			}
+
+		default:
```
Comment (lukanus) [inline extractor.go:186]: I'm quite sure you don't need that

File: extractor.go [hunk 2]
```diff
@@ -405,6 +414,38 @@ func indexBlock(out io.Writer, sync *sync.Mutex, bh types.EventDataNewBlock) err
...
+		return nil
+	}
+
+	result := &codec.EventDataValidatorSetUpdates{}
+
+	for _, update := range updates.ValidatorUpdates {
+		result.ValidatorUpdates = append(result.ValidatorUpdates, &codec.Validator{
+			Address:     update.Address.Bytes(),
+			VotingPower: update.VotingPower,
+			// PubKey: ???,
```
Comment (lukanus) [inline extractor.go:428]: ??? :>

File: extractor.go [hunk 3]
```diff
@@ -405,6 +414,38 @@ func indexBlock(out io.Writer, sync *sync.Mutex, bh types.EventDataNewBlock) err
...
+
+	sync.Lock()
+	defer sync.Unlock()
+
+	data, err := proto.Marshal(result)
+	if err != nil {
+		return fmt.Errorf("cant marshal validator: %v", err)
+	}
+
+	// dans: We need to add height prefix
```
Comment (lukanus) [inline extractor.go:440]: lukasz: yes we need to do that :)

Comment (lavabyrd) [reply extractor.go]: done!

File: extractor.go (hunk 2 above)
Comment (lavabyrd) [reply extractor.go]: Fixed

File: extractor.go (hunk 1 above)
Comment (lavabyrd) [reply extractor.go:152]: removed

File: extractor.go [hunk 4]
```diff
@@ -339,6 +359,56 @@ func indexBlock(out Writer, sync *sync.Mutex, bh types.EventDataNewBlock) error
...
+	defer sync.Unlock()
+
+	return out.WriteLine(fmt.Sprintf("%s %d %s",
+		dmValidator,
+		height,
+		base64.StdEncoding.EncodeToString(data),
+	))
+}
+
+// func formatFilename(name string) string {
```
Comment (sosedoff) [inline extractor.go:404]: You can remove this part, it has been integrated into `Config`

File: extractor.go [hunk 5]
```diff
@@ -383,18 +453,21 @@ func mapVote(edv *types.Vote) *codec.EventDataVote {
 
 func mapValidator(v abci.ValidatorUpdate) (*codec.Validator, error) {
 	nPK := &codec.PublicKey{}
+	var Address []byte
```
Comment (sosedoff) [inline extractor.go:456]: should be lowercase `address`

File: extractor.go (hunk 4 above)
Comment (lavabyrd) [reply extractor.go:404]: done

**Pull request merged**